libminifi/src/Site2SiteClientProtocol.cpp

```diff
@@ -682,6 +682,7 @@ bool Site2SiteClientProtocol::receive(std::string transactionID,
 
 bool Site2SiteClientProtocol::send(
     std::string transactionID, DataPacket *packet, std::shared_ptr<FlowFileRecord> flowFile,
+	uint8_t *payload, int length,
```

Comment:
send can take flowfile or raw bytes, DataPacket is more like a hold for the meta info like transaction, etc.